semgrep-core/src/engine/Match_patterns.mli

```diff
@@ -3,11 +3,20 @@
 (*s: signature [[Semgrep_generic.check]] *)
 val check :
   hook:(Metavariable.bindings -> Parse_info.t list Lazy.t -> unit) ->
+  ?range_filter:(Parse_info.token_location * Parse_info.token_location -> bool) ->
   Config_semgrep.t ->
   Mini_rule.rules ->
   Equivalence.equivalences ->
   Common.filename * Lang.t * Target.t ->
   Pattern_match.t list
+(** Match mini-rules (patterns) against a target file.
+  *
+  * @param range_filter A predicate that defines a range of interest when matching
```

Comment:
I think (given our conventions) this comment is better in the ml. It's more of an implementation doc

Reply:
Do you have a pointer to those conventions? Maybe there was a discussion about this when I had not yet joined R2C full-time? The only thing I remember was actually @mjambon raising the issue that we don't have docs for many interfaces, and I believe we had agreed to document them.

Reply:
It's mostly a sense based on PRs. Reading it again I see why this also is an interface doc. Maybe what I'm expressing is more than when I was looking at the check2 function I thought it would be helpful to have a comment over the range filtering logic to explain why this is useful.

Reply:
Fair enough! I will add a more implementation-oriented comment to `check2`.